File: editorial/chap03/answer_B11.cpp
#include <iostream>
#include <algorithm>
using namespace std;
 
int N, A[100009];
int Q, X[100009];
 
int main() {
	// 入力
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];
	cin >> Q;
	for (int i = 1; i <= Q; i++) cin >> X[i];
 
	// 配列 X をソート
	sort(A + 1, A + N + 1);
 
	// 質問に答える
	for (int i = 1; i <= Q; i++) {
		int pos1 = lower_bound(A + 1, A + N + 1, X[i]) - A;
		cout << pos1 - 1 << endl;
	}
	return 0;
}

File: editorial/chap03/python/answer_B11.py
import bisect

# 入力
N = int(input())
A = list(map(int, input().split()))
Q = int(input())
X = [ None ] * Q
for i in range(Q):
	X[i] = int(input())

# 配列 X をソート
A.sort()

# 質問に答える（配列 A は 0 番目から始まっていることに注意！）
for i in range(Q):
	pos1 = bisect.bisect_left(A, X[i])
	print(pos1)
